<div style="height:1123px; font-family: 'Times New Roman', Times, serif; font-size:19px; text-align:center;"><b>

    LAPORAN PRAKTIKUM

    ALGORITMA & PEMOGRAMAN

    “STATEMENT CONDITIONAL”

    Disusun Oleh :

    SULTAN GHIFARI AL-DJADID

    2611532031

    Dosen Pengampu:

    Wahyudi.DR.. S.T.M.T

    Asisten Praktikum:

    SHERLY SUKMADIRA

<br>

 <img src="images/logo_unand.png" width="200">

<br>

    FAKULTAS TEKNOLOGI INFORMASI
    
    DEPARTEMEN INFORMATIKA

    UNIVERSITAS ANDALAS

    2026

</br>
</div>

---
<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">
<h1 align="center"><b>KATA PENGANTAR</b></h1>

Puji syukur penulis panjatkan atas kehadirat Tuhan Yang Maha Esa yang telah memberikan rahmat dan berkatnya, sehingga penulis dapat menyelesaikan laporan praktikum tentang Statement Conditional yang telah dilaksanakan sebagai pertemuan ke-4 untuk mata kuliah Praktikum Struktur Data. Praktikum ini berfokus Statement Conditional yang digunakan untuk menentukan jenis nilai program If, Elif, dan Else

Melalui praktikum ini, hasil yang diharapkan adalah penulis dapat memahami berbagai jenis Statement Conditional dan cara menggunakan variabel dalam Python serta mampu menerapkannya dalam pembuatan program sederhana. Statement Conditional adalah jenis data yang digunakan dalam program, seperti angka, teks, atau nilai benar dan salah. Statement Conditional membantu komputer mengetahui cara mengolah data tersebut. Variabel adalah tempat untuk menyimpan data dalam program. Data yang disimpan dalam variabel bisa digunakan kembali atau diubah sesuai kebutuhan.

Penulis menyadari bahwa laporan praktikum ini masih memiliki banyak kekurangan. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan demi perbaikan dan pengembangan di masa mendatang.

<div align="right" style="margin-left:-30px;">
Padang, 2026
<br>
<br>
<br>
<br>
<br>
<br>
Sultan Ghifari Al-Djadid
<br>
2611532031
</div>
</div>

---

<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">
<h1 align="center"><b>BAB I <br> PENDAHULUAN</b></h1>

## 1.1 Latar Belakang
<p style="text-indent: 40px; text-align: justify; ">Dalam pemrograman, komputer perlu mampu mengambil keputusan, bukan hanya menjalankan perintah secara berurutan. Untuk itu, digunakanlah conditional statement (pernyataan kondisional), yaitu struktur kontrol yang memungkinkan program menjalankan blok kode tertentu berdasarkan suatu kondisi. Konsep ini terinspirasi dari cara berpikir manusia dalam kehidupan sehari-hari, misalnya: "jika hujan, maka bawa payung; jika tidak, maka tidak perlu." Logika sederhana ini diterapkan dalam program melalui pernyataan if, elif, dan else. Dengan adanya struktur ini, program menjadi lebih fleksibel karena dapat merespons berbagai kemungkinan situasi, seperti validasi input, sistem penilaian, login, hingga logika pada aplikasi yang lebih kompleks. Oleh karena itu, pemahaman terhadap if, elif, dan else menjadi dasar penting dalam belajar pemrograman.
</p>

## 1.2 Tujuan
1. Mengetahui pengertian dan fungsi Statement Conditional.
2. Memahami jenis-jenis Statement Conditional.
3. Mengetahui cara membuat dan menggunakan variabel.
4. Mampu menggunakan Statement Conditional dalam membuat program sederhana.
5. Memahami cara menyimpan dan mengolah data dalam program
## 1.3 Manfaat
1. Mahasiswa memahami dasar-dasar pemrograman dengan lebih mudah.
2. Mahasiswa meningkatkan kemampuan berpikir logis dalam menyelesaikan masalah.
3. Mahasiswa mampu membuat program sederhana menggunakan Python.
4. Mahasiswa memahami cara menyimpan dan mengolah data dalam program.
5. Mahasiswa menjadi bekal untuk mempelajari pemrograman lebih lanjut.
</div>

---
<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
<h1 align="center"><b>BAB II<br>PEMBAHASAN </b> </h1>

## 2.1 Dasar Teori
Pengenalan tentang Statement Conditional


1. Pengertian Conditional Statement<br>
Conditional statement (pernyataan kondisional) adalah struktur kontrol dalam pemrograman yang digunakan untuk menjalankan suatu blok kode tertentu berdasarkan hasil evaluasi sebuah kondisi atau ekspresi logika (boolean). Kondisi tersebut hanya memiliki dua kemungkinan nilai, yaitu True (benar) atau False (salah). Berdasarkan nilai inilah program menentukan alur eksekusi selanjutnya.

2. Konsep Percabangan (Branching)
Struktur conditional sering disebut juga sebagai struktur percabangan, karena program tidak lagi berjalan secara linear (satu alur), melainkan bercabang menjadi beberapa kemungkinan jalur eksekusi, tergantung pada kondisi yang diperiksa. Secara umum terdapat tiga bentuk percabangan:<br>
Percabangan satu kondisi → hanya menggunakan if<br>
Percabangan dua kondisi → menggunakan if dan else<br>
Percabangan banyak kondisi → menggunakan if, elif, dan else

3. Teori Dasar Masing-Masing Pernyataan<br>
a. if<br>
Pernyataan if berfungsi mengevaluasi suatu ekspresi boolean. Apabila hasil evaluasinya bernilai True, maka blok kode yang berada di dalamnya akan dieksekusi. Sebaliknya, jika bernilai False, blok kode tersebut akan dilewati begitu saja.<br>
b. elif<br>
Pernyataan elif (kependekan dari else if) digunakan untuk memeriksa kondisi tambahan apabila kondisi pada if sebelumnya bernilai False. Dalam satu program, elif dapat digunakan lebih dari satu kali untuk menangani banyak kemungkinan kondisi secara berurutan, dan sifatnya eksklusif — artinya hanya satu blok saja yang akan dieksekusi meskipun terdapat banyak elif.<br>
c. else<br>
Pernyataan else tidak memiliki kondisi (ekspresi boolean) dan berfungsi sebagai blok "penutup" yang akan dijalankan apabila seluruh kondisi pada if dan elif di atasnya bernilai False.

4. Alur Eksekusi (Flow of Control)<br>
Secara teori, sistem akan mengevaluasi kondisi-kondisi tersebut dari atas ke bawah secara berurutan, dengan tahapan sebagai berikut:<br>
Kondisi pada if diperiksa terlebih dahulu.<br>
Jika bernilai False, sistem akan melanjutkan pemeriksaan ke elif (jika ada), satu per satu secara berurutan.<br>
Apabila seluruh kondisi pada if maupun elif bernilai False, maka blok else yang akan dijalankan.<br>
Begitu ditemukan satu kondisi yang bernilai True, blok kode terkait akan dieksekusi, dan seluruh percabangan lain yang tersisa tidak akan diperiksa lagi (diabaikan).<br>

5. Operator yang Umum Digunakan dalam Kondisi<br>
Kondisi yang diperiksa pada if maupun elif biasanya dibentuk dengan menggunakan dua jenis operator, yaitu:<br>
Operator perbandingan, seperti sama dengan, tidak sama dengan, lebih besar, lebih kecil, lebih besar atau sama dengan, dan lebih kecil atau sama dengan.<br>
Operator logika, seperti and (dan), or (atau), dan not (bukan/negasi), yang digunakan untuk menggabungkan atau membalik beberapa kondisi sekaligus
.
6. Sifat Penting Conditional Statement<br>
Bersifat eksklusif mutual, artinya hanya satu blok kode yang akan dijalankan dari keseluruhan struktur if-elif-else.<br>
Urutan pemeriksaan kondisi sangat berpengaruh terhadap hasil akhir program, sehingga penyusunan urutan kondisi perlu dilakukan secara cermat dan logis.<br>
Struktur ini menjadi dasar penting sebelum mempelajari konsep pemrograman lanjutan lainnya, seperti perulangan (looping) dan fungsi (function).
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

## 2.2 Kode Program

### 2.2.1 Program "IF1_2611532031"
</div>

In [ ]:
#Buat file dengan nama If1_2611532031.py
#Buat program untuk kondisional if
#Nama variabel ditambah 4 digit nim terakhir contoh: ipk_2031
#Program ini menggunakan fungsi input()

IPK_2031 = float(input("Input IPK Anda = "))

if IPK_2031 > 2.75:
    print("Anda Lulus Sangat Memuaskan dengan IPK " + str(IPK_2031))
    
print("Program Selesai")

Anda lulus sangat memuaskan dengan IPK4.0
Program Selesai


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

Penjelasan:
1. Input Data<br>
Program meminta pengguna memasukkan nilai IPK melalui fungsi input().<br>
Karena hasil dari input() selalu berupa teks (string), nilai tersebut langsung diubah menjadi tipe data desimal (float) agar bisa dibandingkan secara matematis, bukan dibandingkan sebagai teks.<br><br>
2. Struktur Percabangan (if tunggal)<br>
Program menggunakan satu kondisi if saja, tanpa pasangan else.<br>
Kondisi yang dicek adalah apakah IPK yang diinput lebih besar dari 2.75.<br>
Jika IPK yang dimasukkan lebih besar dari 2.75, maka kondisi bernilai benar, sehingga program mencetak pesan kelulusan yang menyatakan pengguna lulus dengan predikat sangat memuaskan beserta nilai IPK-nya.<br>
Jika IPK yang dimasukkan kurang dari atau sama dengan 2.75, maka kondisi bernilai salah, sehingga pesan kelulusan tersebut tidak dicetak sama sekali.<br>
Karena tidak ada else sebagai penangkap kondisi sebaliknya, tidak ada pesan alternatif apa pun yang muncul untuk kasus ini.<br><br>
Pada bagian pencetakan pesan, nilai IPK yang bertipe angka digabungkan dengan teks menggunakan operator penggabungan string.<br>
Karena angka tidak bisa langsung digabung dengan teks di Python, nilai IPK tersebut terlebih dahulu diubah kembali menjadi bentuk teks sebelum digabungkan — proses ini disebut type casting (konversi tipe data).<br><br>
3. Penutup Program<br>
Di baris terakhir, program mencetak pesan "Program Selesai".<br>
Pesan ini selalu muncul, apa pun hasil dari pengecekan IPK sebelumnya — baik IPK-nya memenuhi syarat maupun tidak — karena baris ini berada di luar struktur if, sehingga dieksekusi secara terpisah setelah blok kondisi selesai diproses.<br><br>
Contoh Skenario:<br>
Jika pengguna memasukkan IPK 3.5 (lebih besar dari 2.75), maka akan muncul pesan kelulusan sangat memuaskan, diikuti pesan "Program Selesai".<br>
Jika pengguna memasukkan IPK 2.5 (lebih kecil dari 2.75), maka pesan kelulusan tidak akan muncul, dan program langsung menampilkan "Program Selesai" saja.<br><br>
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

### 2.2.2 Program "IF2_2611532031"
</div>

In [ ]:
#Buat file dengan nama If2_2611532031.py
#Buat program untuk kondisional if
#Nama variabel ditambah 4 digit nim terakhir contoh: ipk_2031
#Program ini menggunakan fungsi input()

IPK_2031 = float(input("Input IPK_2031 Anda = "))

if IPK_2031 > 2.75:
    print("Anda Lulus Sangat Memuaskan dengan IPK " + str(IPK_2031))
else:
    print("Anda Tidak Lulus")
print("Program Selesai")

Anda lulus sangat memuaskan dengan IPK4.0


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data<br>
Program meminta pengguna memasukkan nilai IPK melalui fungsi input().<br>
Karena hasil dari input() selalu berupa teks (string), nilai tersebut langsung diubah menjadi tipe data desimal (float) agar bisa dibandingkan secara matematis, bukan dibandingkan sebagai teks.<br><br>
2. Struktur Percabangan (if-else)<br>
Program menggunakan struktur if-else, yaitu dua kemungkinan hasil yang saling melengkapi (jika kondisi benar dijalankan satu blok, jika salah dijalankan blok yang lain).<br>
Kondisi yang dicek adalah apakah IPK yang diinput lebih besar dari 2.75.<br>
Jika IPK yang dimasukkan lebih besar dari 2.75, maka kondisi if bernilai benar, sehingga program mencetak pesan kelulusan yang menyatakan pengguna lulus dengan predikat sangat memuaskan beserta nilai IPK-nya.<br>
Jika IPK yang dimasukkan kurang dari atau sama dengan 2.75, maka kondisi if bernilai salah, sehingga program otomatis menjalankan blok else, yaitu mencetak pesan "Anda Tidak Lulus".<br>
Berbeda dengan struktur if tunggal, pada if-else ini selalu ada satu pesan yang tercetak — tidak ada kemungkinan program tidak mencetak apa pun, karena setiap kondisi sudah memiliki pasangan penanganannya masing-masing (benar → blok if, salah → blok else).<br><br>
Pada bagian pencetakan pesan kelulusan, nilai IPK yang bertipe angka digabungkan dengan teks menggunakan operator penggabungan string.<br>
Karena angka tidak bisa langsung digabung dengan teks di Python, nilai IPK tersebut terlebih dahulu diubah kembali menjadi bentuk teks sebelum digabungkan — proses ini disebut type casting (konversi tipe data).<br>
Pesan pada blok else ("Anda Tidak Lulus") tidak menyertakan nilai IPK, sehingga tidak memerlukan proses penggabungan atau konversi tipe data apa pun.<br><br>
Contoh Skenario:<br>
Jika pengguna memasukkan IPK 3.5 (lebih besar dari 2.75), maka kondisi if terpenuhi dan akan muncul pesan kelulusan sangat memuaskan beserta nilai IPK-nya.<br>
Jika pengguna memasukkan IPK 2.5 (lebih kecil dari 2.75), maka kondisi if tidak terpenuhi, sehingga program menjalankan blok else dan mencetak pesan "Anda Tidak Lulus".<br><br>
Poin Penting Struktur if-else:<br>
Ciri khas if-else adalah ia selalu menangani dua kemungkinan sekaligus — kondisi benar dan kondisi salah — sehingga program pasti akan mencetak salah satu dari dua pesan yang tersedia, tergantung hasil evaluasi kondisinya.<br>
Ini berbeda dengan if tunggal yang hanya menangani satu kemungkinan (benar) dan membiarkan kondisi sebaliknya tanpa penanganan apa pun.
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

### 2.2.3 Program "MULTI_IF1_2611532031
</div>

In [ ]:
#Buat file dengan nama Multi_If1_2611532031.py
#Buat program untuk kondisional if
#Nama variabel ditambah 4 digit nim terakhir contoh: ipk_2031
#Program ini menggunakan fungsi input()

Umur_2031 = int(input("Input Umur Anda: "))
Sim_2031 = input("Apakah Anda Sudah Punya SIM C (y/t): ")[0].lower()

if Umur_2031 >= 17 and Sim_2031 == 'y':
    print("Anda Sudah Dewasa dan Boleh Bawa Motor")

elif Umur_2031 >= 17 and Sim_2031 != 'y':
    print("Anda Sudah Dewasa tetapi Tidak Boleh Bawa Motor")

elif Umur_2031 <= 17 and Sim_2031 == 'y':
    print("Anda Belum Cukup Umur untuk Punya SIM")


else :
    print("Anda Belum Cukup Umur dan Tidak Boleh Bawa Motor")
print("Program Selesai")

Anda sudah dewasa dan boleh bawa motor


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data<br>
Program meminta pengguna memasukkan umur melalui fungsi input(), lalu hasilnya langsung diubah menjadi tipe data bilangan bulat (integer) agar bisa dibandingkan secara matematis.<br>
Program juga meminta status kepemilikan SIM C dengan jawaban "y" (ya) atau "t" (tidak).<br>
Dari jawaban tersebut, hanya karakter pertamanya saja yang diambil menggunakan pengindeksan string [0], lalu langsung disimpan ke dalam variabel — sehingga meskipun pengguna mengetik "ya" atau "yes", yang tersimpan tetap huruf pertamanya saja, yaitu "y".<br><br>
2. Struktur Percabangan (Multi-if Terpisah dengan Operator Logika)<br>
Program ini menggunakan empat blok if yang berdiri sendiri-sendiri (bukan if-elif-else berantai), sehingga setiap blok dievaluasi secara independen satu per satu tanpa saling menggugurkan.<br>
Setiap kondisi menggabungkan dua syarat sekaligus menggunakan operator logika and, yaitu kombinasi antara umur dan status kepemilikan SIM C.<br><br>
Blok pertama mengecek apakah umur lebih besar sama dengan 17 dan sudah punya SIM C.<br>
Jika kedua syarat tersebut terpenuhi bersamaan, maka dicetak pesan bahwa pengguna sudah dewasa dan boleh membawa motor.<br><br>
Blok kedua mengecek apakah umur lebih besar sama dengan 17 tetapi belum/tidak punya SIM C (menggunakan operator tidak sama dengan).<br>
Jika kedua syarat tersebut terpenuhi bersamaan, maka dicetak pesan bahwa pengguna sudah dewasa tetapi tidak boleh membawa motor karena belum memiliki SIM.<br><br>
Blok ketiga mengecek apakah umur kurang dari 17 tetapi sudah punya SIM C.<br>
Jika kedua syarat tersebut terpenuhi bersamaan, maka dicetak pesan bahwa pengguna belum cukup umur namun sudah memiliki SIM — kasus ini menandakan kejanggalan data karena secara aturan seharusnya SIM baru bisa dimiliki di atas umur tertentu.<br><br>
Blok keempat mengecek apakah umur kurang dari 17 dan juga belum/tidak punya SIM C.<br>
Jika kedua syarat tersebut terpenuhi bersamaan, maka dicetak pesan bahwa pengguna belum cukup umur untuk membawa motor.<br><br>
3. Karakteristik Penting Multi-if Terpisah<br>
Karena keempat blok ini ditulis sebagai if yang terpisah (bukan elif), Python akan tetap mengevaluasi seluruh empat kondisi tersebut satu per satu dari atas ke bawah, meskipun salah satu kondisi di atasnya sudah terpenuhi.<br>
Namun, karena keempat kondisi tersebut saling eksklusif (tidak mungkin dua kondisi sekaligus bernilai benar untuk kombinasi umur dan SIM yang sama), maka pada praktiknya hanya satu pesan saja yang akan tercetak untuk setiap kali program dijalankan.<br>
Berbeda dengan struktur if-elif-else yang otomatis berhenti begitu satu kondisi terpenuhi, struktur multi-if ini tetap memeriksa semua kondisi hingga blok terakhir, sehingga cocok digunakan saat beberapa hasil berpotensi ingin diakumulasikan (meski dalam kasus ini hasilnya tetap saling eksklusif).<br><br>
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

### 2.2.4 Program "MULTI_IF2_2611532031
</div>

In [ ]:
#Buat file dengan nama Multi_If1_2611532031.py
#Buat program untuk kondisional if
#Nama variabel ditambah 4 digit nim terakhir contoh: ipk_2031
#Program ini menggunakan fungsi input()
#Program Menghitung Diskon Belanja

#Input dari user
Total_Belanja_2031 = float(input("Masukkan Total Belanja (Rp): "))

#Input status member (mengecek apakah user mengetik 'y' atau 'ya')
Input_Member_2031 = input("Apakah Anda Member? (y/ya): ").strip().lower()
Is_Member_2031 = Input_Member_2031 in ["y", "ya"]

#Input status kode promo (mengecek apakah user mengetik 'y' atau 'ya')
Input_Promo_2031 = input("Apakah Kode Promo Valid? (y/ya): ").strip().lower()
Kode_Promo_2031 = Input_Promo_2031 in ["y", "ya"]

Total_Diskon_Persen = 0

#Multi-IF terpisah: Setiap kondisi diperiksa secara Independen
#Diskon Bisa Ditumpuk (akumulasi) jika memenuhi beberapa syarat sekaligus

if Total_Belanja_2031 > 1000000: 
    Total_Diskon_Persen += 10 # Diskon Belanja Besar

if Is_Member_2031:
    Total_Diskon_Persen += 5 # Diskon Member

if Kode_Promo_2031:
    Total_Diskon_Persen += 15 # Diskon Voucher

#Menghitung nominal diskon dan total bayar
Nominal_Diskon_2031 = Total_Belanja_2031 * (Total_Diskon_Persen / 100)
Total_Bayar_2031 = Total_Belanja_2031 - Nominal_Diskon_2031

#Output hasil 
print("\n--- Rincian Pembayaran ---")
print(f"Total Diskon : {Total_Diskon_Persen}% (Rp {Nominal_Diskon_2031:,.0F})")
print(f"Total Bayar : Rp {Total_Bayar_2031:,.0F}")

print(f"Total Diskon yang Anda Dapatkan: {Total_Diskon_Persen}%")
#Output Total Diskon yang Anda Dapatkan: 30% jika belanja > 1 juta, member, dan kode promo valid


--- Rincian Pembayaran---
Total Diskon : 30% (Rp1,500,000)
Total_bayar_3011: Rp 3,500,000
Total diskon yang anda dapatkan: 30%


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data<br>
Program meminta pengguna memasukkan total belanja melalui fungsi input(), lalu hasilnya langsung diubah menjadi tipe data desimal (float) agar bisa digunakan dalam perhitungan matematis.<br>
Program juga meminta status keanggotaan member dengan jawaban "y" atau "t", di mana teks yang diinput dirapikan terlebih dahulu menggunakan strip() (menghapus spasi berlebih) dan lower() (mengubah menjadi huruf kecil), sehingga input seperti " Y " atau "YA" tetap bisa terbaca dengan benar.<br>
Hasil jawaban tersebut kemudian dicek apakah termasuk dalam daftar ["y", "ya"], dan hasil pengecekan ini (bernilai benar atau salah) langsung disimpan ke dalam sebuah variabel bertipe boolean.<br>
Proses yang sama juga dilakukan untuk input status kode promo, sehingga didapat variabel boolean lain yang menandakan apakah kode promo yang dimasukkan valid atau tidak.<br><br>
2. Struktur Percabangan (Multi-if Terpisah untuk Akumulasi Diskon)<br>
Program ini menggunakan tiga blok if yang berdiri sendiri-sendiri (bukan elif), sehingga setiap kondisi dievaluasi secara independen dan diskonnya bisa ditumpuk atau diakumulasikan, bukan saling menggugurkan satu sama lain.<br><br>
Blok pertama mengecek apakah total belanja lebih besar dari Rp 1.000.000.<br>
Jika kondisi ini terpenuhi, maka persentase diskon ditambahkan sebesar 10% sebagai diskon belanja besar.<br><br>
Blok kedua mengecek apakah pengguna berstatus member (berdasarkan variabel boolean yang telah dihasilkan sebelumnya).<br>
Jika kondisi ini terpenuhi, maka persentase diskon ditambahkan sebesar 5% sebagai diskon member.<br><br>
Blok ketiga mengecek apakah kode promo yang dimasukkan valid (berdasarkan variabel boolean kode promo).<br>
Jika kondisi ini terpenuhi, maka persentase diskon ditambahkan sebesar 15% sebagai diskon promo.<br><br>
Karena ketiga blok tersebut ditulis sebagai if yang terpisah, seluruh kondisi akan tetap dicek satu per satu meskipun kondisi sebelumnya sudah terpenuhi, sehingga jika pengguna memenuhi ketiga syarat sekaligus (belanja di atas Rp 1.000.000, berstatus member, dan memiliki kode promo valid), maka ketiga persentase diskon tersebut akan dijumlahkan seluruhnya menjadi satu total diskon akhir.<br><br>
3. Perhitungan Nominal Diskon dan Total Bayar<br>
Setelah total persentase diskon terkumpul dari ketiga blok if di atas, nominal diskon dihitung dengan mengalikan total belanja dengan persentase diskon tersebut (dibagi 100 untuk mengubahnya dari bentuk persen ke bentuk desimal).<br>
Total bayar akhir kemudian dihitung dengan mengurangkan total belanja dengan nominal diskon yang telah didapat.<br><br>
4. Output Hasil<br>
Program mencetak rincian pembayaran menggunakan f-string dengan format angka :,.0f, yang berfungsi menampilkan angka dengan pemisah ribuan (koma) dan tanpa angka di belakang koma (dibulatkan menjadi bilangan bulat), sehingga tampilannya lebih mudah dibaca layaknya format mata uang.<br>
Ditampilkan total persentase diskon beserta nominal potongannya dalam rupiah, total yang harus dibayar setelah dipotong diskon, dan diulang kembali persentase total diskon yang berhasil didapatkan pengguna sebagai ringkasan penutup.<br><br>
Contoh Skenario:<br>
Jika pengguna berbelanja senilai Rp 5.000.000, berstatus member, dan memiliki kode promo valid, maka ketiga kondisi if terpenuhi sekaligus, sehingga total diskon terakumulasi menjadi 10% + 5% + 15% = 30%, dan total bayar dihitung berdasarkan potongan 30% tersebut dari total belanja.<br>
Jika pengguna berbelanja senilai Rp 500.000, bukan member, dan tidak memiliki kode promo, maka tidak ada satu pun kondisi if yang terpenuhi, sehingga total diskon tetap 0% dan total bayar sama dengan total belanja awal.<br><br>
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

### 2.2.5 Program NILAI_2611532031
</div>

In [ ]:
#Buat file dengan nama Nilai_2611532031.py
#Buat program untuk nested if
#Nama variabel ditambah 4 digit nim terakhir contoh: ipk_2031
#Program ini menggunakan fungsi input()
#Program konversi nilai angka menjadi huruf

Nilai_2031 = int(input("Inputkan Nilai Angka= "))

if Nilai_2031 >= 81:
    print("A")
elif Nilai_2031 >= 70:
    print("B")
elif Nilai_2031 >= 60:
    print("C")
elif Nilai_2031 >= 50:
    print("D")
else:
    print("E")

A


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data<br>
Program meminta pengguna memasukkan nilai angka melalui fungsi input(), lalu hasilnya langsung diubah menjadi tipe data bilangan bulat (integer) agar bisa dibandingkan secara matematis dengan batas-batas rentang nilai yang telah ditentukan.<br><br>
2. Struktur Percabangan (if-elif-else Berantai)<br>
Program ini menggunakan satu rangkaian kondisi if-elif-elif-elif-else yang saling berurutan dan bertingkat satu level (sejajar), di mana Python akan mengecek kondisi dari atas ke bawah dan langsung berhenti begitu menemukan kondisi pertama yang bernilai benar, tanpa memeriksa kondisi-kondisi selanjutnya.<br><br>
Kondisi pertama mengecek apakah nilai yang diinput lebih besar sama dengan 81.<br>
Jika terpenuhi, program mencetak huruf "A" sebagai predikat nilai tertinggi, lalu langsung berhenti mengevaluasi kondisi lainnya.<br><br>
Kondisi kedua (elif) hanya akan dicek jika kondisi pertama tidak terpenuhi (nilai kurang dari 81), yaitu mengecek apakah nilai lebih besar sama dengan 70.<br>
Jika terpenuhi, program mencetak huruf "B".<br>
Perhatikan bahwa karena kondisi ini baru dicek setelah kondisi pertama gagal, maka secara otomatis nilai yang masuk ke sini sudah pasti berada di rentang 70 sampai 80.<br><br>
Kondisi ketiga (elif) hanya dicek jika dua kondisi sebelumnya tidak terpenuhi, yaitu mengecek apakah nilai lebih besar sama dengan 60.<br>
Jika terpenuhi, program mencetak huruf "C", yang berarti nilai berada di rentang 60 sampai 69.<br><br>
Kondisi keempat (elif) hanya dicek jika ketiga kondisi sebelumnya tidak terpenuhi, yaitu mengecek apakah nilai lebih besar sama dengan 50.<br>
Jika terpenuhi, program mencetak huruf "D", yang berarti nilai berada di rentang 50 sampai 59.<br><br>
Kondisi terakhir (else) akan otomatis dijalankan jika seluruh kondisi di atas tidak ada satupun yang terpenuhi, artinya nilai yang diinput berada di bawah 50.<br>
Dalam kasus ini, program mencetak huruf "E" sebagai predikat nilai terendah.<br><br>
3. Karakteristik Penting Struktur if-elif-else<br>
Berbeda dengan multi-if terpisah yang mengecek semua kondisi satu per satu tanpa saling menggugurkan, struktur if-elif-else ini bersifat eksklusif — begitu satu kondisi terpenuhi dan blok di dalamnya dijalankan, seluruh kondisi elif dan else di bawahnya otomatis dilewati tanpa dicek lagi.<br>
Hal ini membuat struktur ini sangat cocok digunakan untuk kasus pengelompokan rentang nilai seperti konversi nilai angka ke huruf, karena setiap angka hanya boleh masuk ke satu kategori huruf saja, tidak boleh lebih dari satu.<br><br>
Contoh Skenario:<br>
Jika pengguna memasukkan nilai 85, maka kondisi pertama (≥81) langsung terpenuhi, sehingga tercetak huruf "A" dan seluruh kondisi elif/else di bawahnya tidak diperiksa lagi.<br>
Jika pengguna memasukkan nilai 45, maka seluruh kondisi if dan elif tidak ada yang terpenuhi, sehingga program menjalankan blok else dan mencetak huruf "E".
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

### 2.2.6 Program IF_ELIF_ELSE1_2611532031
</div>

In [7]:
#Buat file dengan nama If_Elif_Else1_2611532031.py
#Buat program untuk kondisional if
#Nama variabel ditambah 4 digit nim terakhir contoh: ipk_2031
#Program ini menggunakan fungsi input()

Umur_2031 = int(input("Input Umur Anda: "))
Sim_2031 = input("Apakah Anda sudah punya Sim C (y/t): ")[0].lower()

if Umur_2031 >= 17 and Sim_2031 == "y":
    print("Anda sudah dewasa dan Boleh Bawa Motor")

elif Umur_2031 >= 17 and Sim_2031 != "y":
    print("Anda Sudah Dewasa tetapi Tidak Boleh Bawa Motor")

elif Umur_2031 < 17 and Sim_2031 =="y":
    print("Anda Belum Cukup Umur dan Punya SIM")

else:
    print("Anda Belum Cukup Umur dan Tidak Boleh Bawa Motor")
print("Program Selesai")

Anda sudah dewasa dan boleh bawa motor
Program selesai


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data<br>
Program meminta pengguna memasukkan umur melalui fungsi input(), lalu hasilnya langsung diubah menjadi tipe data bilangan bulat (integer) agar bisa dibandingkan secara matematis.<br>
Program juga meminta status kepemilikan SIM C dengan jawaban "y" (ya) atau "t" (tidak).<br>
Dari jawaban tersebut, hanya karakter pertamanya saja yang diambil menggunakan pengindeksan string [0], sehingga meskipun pengguna mengetik "ya" atau "yes", yang tersimpan tetap huruf pertamanya saja, yaitu "y".<br><br>
2. Struktur Percabangan (if-elif-else dengan Operator Logika)<br>
Program ini menggunakan satu rangkaian kondisi if-elif-elif-else yang saling berurutan dan sejajar (satu level, bukan nested), di mana Python akan mengecek kondisi dari atas ke bawah dan langsung berhenti begitu menemukan kondisi pertama yang bernilai benar, tanpa memeriksa kondisi-kondisi selanjutnya.<br>
Setiap kondisi menggabungkan dua syarat sekaligus menggunakan operator logika and, yaitu kombinasi antara umur dan status kepemilikan SIM C.<br><br>
Kondisi pertama mengecek apakah umur lebih besar sama dengan 17 dan sudah punya SIM C ("y") secara bersamaan.<br>
Jika kedua syarat tersebut terpenuhi, program mencetak pesan bahwa pengguna sudah dewasa dan boleh membawa motor, lalu langsung berhenti mengevaluasi kondisi lainnya.<br><br>
Kondisi kedua (elif) hanya dicek jika kondisi pertama tidak terpenuhi, yaitu mengecek apakah umur lebih besar sama dengan 17 tetapi belum/tidak punya SIM C (menggunakan operator tidak sama dengan).<br>
Jika kedua syarat tersebut terpenuhi, program mencetak pesan bahwa pengguna sudah dewasa tetapi tidak boleh membawa motor karena belum memiliki SIM.<br><br>
Kondisi ketiga (elif) hanya dicek jika kedua kondisi sebelumnya tidak terpenuhi, yaitu mengecek apakah umur kurang dari 17 tetapi sudah punya SIM C.<br>
Jika kedua syarat tersebut terpenuhi, program mencetak pesan bahwa pengguna belum cukup umur namun sudah memiliki SIM — kasus ini menandakan kejanggalan data karena secara aturan seharusnya SIM baru bisa dimiliki di atas umur tertentu.<br><br>
Kondisi terakhir (else) akan otomatis dijalankan jika ketiga kondisi di atas tidak ada satupun yang terpenuhi, artinya umur di bawah 17 dan juga belum/tidak punya SIM C.<br>
Dalam kasus ini, program mencetak pesan bahwa pengguna belum cukup umur dan tidak boleh membawa motor.<br><br>
3. Penutup Program<br>
Di baris terakhir, program mencetak pesan "Program selesai".<br>
Pesan ini selalu muncul, apa pun hasil dari keempat kondisi di atas, karena baris ini berada di luar struktur if-elif-else, sehingga dieksekusi secara terpisah setelah salah satu blok kondisi selesai diproses.<br><br>
Jika pengguna memasukkan umur 20 dan menjawab "y" untuk kepemilikan SIM, maka kondisi pertama langsung terpenuhi, sehingga tercetak pesan bahwa pengguna sudah dewasa dan boleh membawa motor, diikuti pesan "Program selesai". Jika pengguna memasukkan umur 15 dan menjawab "t" untuk kepemilikan SIM, maka ketiga kondisi if dan elif tidak ada yang terpenuhi, sehingga program menjalankan blok else dan mencetak pesan bahwa pengguna belum cukup umur dan tidak boleh membawa motor, diikuti pesan "Program selesai".
</div>

<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

### 2.2.7 Program MACTH_CASE1_2611532031
</div>

In [ ]:
#Buat file dengan nama Match_Case1_2611532031.py
#Buat program untuk match case
#Nama variabel ditambah 4 digit nim terakhir contoh: ipk_2031
#Program ini menggunakan fungsi input()
#Program konversi angka menjadi nama bulan

Bulan_2031 = int(input("Masukkan Angka Bulan (1 - 12): " ))

match Bulan_2031:
    case 1:
        print("Januari")
    case 2:
        print("Februari")
    case 3:
        print("Maret")
    case 4:
        print("April")
    case 5:
        print("Mei")
    case 6:
        print("Juni")
    case 7:
        print("Juli")
    case 8:
        print("Agustus")
    case 9:
        print("September")
    case 10:
        print("November")
    case 11:
        print("Oktober")
    case 12:
        print("Desember")
    case _:
        print("Angka Tidak Valid")

Juni


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data<br>
Program meminta pengguna memasukkan angka bulan (1 sampai 12) melalui fungsi input(), lalu hasilnya langsung diubah menjadi tipe data bilangan bulat (integer) agar bisa dicocokkan dengan nilai-nilai angka pada struktur match-case.<br><br>
2. Struktur Percabangan (match-case)<br>
Program ini menggunakan struktur match-case, yaitu bentuk percabangan yang mencocokkan satu nilai variabel dengan beberapa kemungkinan pola (case) yang telah ditentukan, mirip seperti if-elif-else namun dengan penulisan yang lebih ringkas dan rapi untuk kasus pencocokan nilai tunggal.<br>
Nilai yang diinput oleh pengguna dicocokkan satu per satu terhadap dua belas case yang tersedia, masing-masing mewakili angka 1 sampai 12.<br><br>
Jika angka yang diinput cocok dengan salah satu case, misalnya case 1, maka program akan mencetak nama bulan yang sesuai, yaitu "Januari".<br>
Begitu pula seterusnya, case 2 akan mencetak "Februari", case 3 mencetak "Maret", hingga case 12 yang mencetak "Desember" — setiap case mewakili satu bulan yang berurutan sesuai kalender.<br><br>
Begitu satu case yang cocok ditemukan, Python akan langsung menjalankan blok print() di dalamnya dan tidak lagi memeriksa case-case lain di bawahnya, sehingga hanya satu nama bulan saja yang akan tercetak untuk setiap kali program dijalankan.<br><br>
Case terakhir, yaitu case _, berfungsi sebagai case default atau penangkap semua kemungkinan nilai yang tidak cocok dengan case manapun di atasnya (angka 1 sampai 12).<br>
Jika pengguna memasukkan angka di luar rentang 1-12, misalnya 0, 13, atau angka negatif, maka program akan otomatis masuk ke case ini dan mencetak pesan "Angka tidak valid" sebagai penanda bahwa input yang dimasukkan salah.<br><br>
3. Karakteristik Penting Struktur match-case<br>
Struktur match-case sangat cocok digunakan ketika kita perlu mencocokkan satu variabel dengan banyak kemungkinan nilai spesifik (seperti pada kasus ini, dua belas kemungkinan angka bulan), karena penulisannya lebih terstruktur dan mudah dibaca dibandingkan menuliskan dua belas kondisi if-elif secara berurutan.<br>
Case _ yang berperan sebagai default sangat penting untuk selalu disertakan di bagian akhir, agar program tetap bisa menangani input yang tidak terduga atau di luar rentang yang valid, sehingga program tidak berhenti secara tiba-tiba (error) maupun tidak mencetak apa pun sama sekali.<br><br>
Contoh Skenario:<br>
Jika pengguna memasukkan angka 6, maka program akan mencocokkan dengan case 6 dan mencetak "Juni".<br>
Jika pengguna memasukkan angka 15 (di luar rentang 1-12), maka tidak ada satu case pun yang cocok, sehingga program masuk ke case _ dan mencetak pesan "Angka tidak valid".<br><br>
</div>

---
<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">

## TUGAS PRAKTIKUM

**Soal :**

<div align="center">
<img src="images/Tugas_pekan4.png" width="500"><br>
<em>Gambar 2.2 : Soal Tugas Praktikum</em>
</div>

**Kode Jawaban :**

a. Kode Program tugas4_2611532031

In [ ]:
import sys
print("=== SISTEM LOKET ALPRO ADVENTURE PARK ===")

#===========================================
# ----------INPUT DATA PENGUNJUNG----------
#===========================================

Nama_2031 =     input("Masukkan Nama Pengunjung            : ")
Umur_2031 = int(input("Input Umur Anda                     : "))
if Umur_2031 <= 9:
    print("Status Akses: Anda Belum Cukup Umur")
    exit()

Sim_2031  =     input("Apakah Anda Sudah Punya SIM C (y/t) : ").strip().lower()[0]

print()
print("Pilihan Paket Wahana (1-5):")
print("  1. Safari Rimba         (Rp 50,000)")
print("  2. Arung Jeram          (Rp 75,000)")
print("  3. Motor ATV Ekstrim    (Rp 120,000)")
print("  4. Roller Coaster Kilat (Rp 100,000)")
print("  5. All-Access VIP       (Rp 220,000)")

Paket_2031 = int(input("Masukkan Nomor Paket (1-5)     : "))


#===================================================
# ----------PEMILIHAN WAHANA (Match-Case)----------
#===================================================

Nama_Paket_2031 = ""
Harga_Satuan_2031 = 0

match Paket_2031:
    case 1:
        Nama_Paket_2031 = "Wahana Safari Rimba"
        Harga_Satuan_2031 = 50000
    case 2:
        Nama_Paket_2031 = "Wahana Arung Jeram"
        Harga_Satuan_2031 = 75000
    case 3:
        Nama_Paket_2031 = "Wahana Motor ATV Ekstrim"
        Harga_Satuan_2031 = 120000
    case 4:
        Nama_Paket_2031 = "Wahana Roller Coaster Kilat"
        Harga_Satuan_2031 = 100000
    case 5:
        Nama_Paket_2031 = "Wahana All-Access VIP"
        Harga_Satuan_2031 = 220000
    case _:
        print("Paket Wahana Tidak Valid!")
        sys.exit()

Jumlah_Tiket_2031 = int(input("Masukkan Jumlah Tiket          : "))
 
if Jumlah_Tiket_2031 <= 0:
    print("Peringatan: Kuota Tiket Tidak Valid!")

Is_Member_2031 =        input("Apakah Anda Member? (y/t)      : ").strip().lower()
Kode_Promo_Valid_2031 = input("Apakah Kode Promo Valid? (y/t) : ").strip().lower()

#================================================
# ----------VALIDASI IZIN KENDALI WAHANA--------
#================================================

print()
print("--- KELAYAKAN PENGENDARA WAHANA ---")

if Paket_2031 == 3 and Umur_2031 >= 17 and Sim_2031 == 'y':
    print("Status Akses: Anda Sudah Dewasa dan Boleh Mengendarai ATV Sendiri.")
elif Paket_2031 == 3 and Umur_2031 >= 17 and Sim_2031 != 'y':
    print("Status Akses: Anda sudah Dewasa tetapi Tidak Boleh Bawa Motor ATV "
          "(Wajib Didampingi Instruktur).")
elif Paket_2031 == 3 and Umur_2031 < 17 and Sim_2031 == 'y':
    print("Status Akses: Identitas Tidak Valid: Belum Cukup Umur Memiliki SIM.")
elif Paket_2031 == 3:
    print("Status Akses: Anda Belum Cukup Umur dan Tidak Boleh Bawa Motor ATV.")
elif Paket_2031 != 3 and Umur_2031 >= 10:
    print(f"Status Akses: Anda Memenuhi Syarat Umur Untuk {Nama_Paket_2031}.")
else:
    print(f"Status Akses: Anda Belum Memenuhi Syarat Umur Untuk {Nama_Paket_2031}.")

#====================================================================
# ----------AKUMULASI DISKON BERTINGKAT (Multi-IF Terpisah)----------
#====================================================================

Subtotal_2031 = Harga_Satuan_2031 * Jumlah_Tiket_2031
Total_Diskon_Persen_2031 = 0

if Subtotal_2031 >= 200000:
    Total_Diskon_Persen_2031 += 10  # Diskon Belanja Besar

if Is_Member_2031 in ['y', 'ya']:
    Total_Diskon_Persen_2031 += 5  # Diskon Member

if Kode_Promo_Valid_2031 in ['y', 'ya']:
    Total_Diskon_Persen_2031 += 15  # Diskon Voucher Promo

if Jumlah_Tiket_2031 >= 5:
    Total_Diskon_Persen_2031 += 5  # Diskon Tambahan Rombongan

#==================================================
# ----------EVALUASI KELULUSAN AUDIT---------------
#==================================================

Nominal_Diskon_2031 = Subtotal_2031 * (Total_Diskon_Persen_2031 / 100)
Total_Bayar_2031 = Subtotal_2031 - Nominal_Diskon_2031

if Total_Bayar_2031 > 300000:
    Catatan_Layanan_2031 = "Selamat! Anda berhak Mendapatkan Souvenir Gratis."
else:
    Catatan_Layanan_2031 = "Terima Kasih Telah Berkunjung."

print()
print("--- Rincian Pembayaran ---")
print(f"Subtotal Belanja : Rp {Subtotal_2031:,.0f}")
print(f"Total Diskon     : {Total_Diskon_Persen_2031}% (Rp {Nominal_Diskon_2031:,.0f})")
print(f"Total Bayar      : Rp {Total_Bayar_2031:,.0f}")
print(f"Catatan Layanan  : {Catatan_Layanan_2031}")
print("Program Selesai")

=== SISTEM LOKET ALPRO ADVENTURE PARK ===

Pilihan Paket Wahana (1-5):
  1. Safari Rimba         (Rp 50,000)
  2. Arung Jeram          (Rp 75,000)
  3. Motor ATV Ekstrim    (Rp 120,000)
  4. Roller Coaster Kilat (Rp 100,000)
  5. All-Access VIP       (Rp 220,000)

--- KELAYAKAN PENGENDARA WAHANA ---
Status Akses: Anda sudah dewasa dan boleh mengendarai ATV sendiri.

--- Rincian Pembayaran ---
Subtotal Belanja : Rp 600,000
Total Diskon     : 35% (Rp 210,000)
Total Bayar      : Rp 390,000
Catatan Layanan  : Selamat! Anda berhak mendapatkan Souvenir Gratis.
Program Selesai


<div style="font-family:'Times New Roman', Times, serif; font-size: 18px;">
Penjelasan:

1. Input Data Awal & Validasi Umur Dasar (if tunggal)<br>
Program dimulai dengan mencetak judul "SISTEM LOKET ALPHA ADVENTURE PARK". Kemudian meminta nama pengunjung, umur, dan status kepemilikan SIM C. Setelah umur diinput, langsung dicek dengan satu kondisi if tunggal: jika umur ≤ 9 tahun, program mencetak pesan "Anda belum cukup umur" dan langsung berhenti (exit()) karena dianggap tidak memenuhi syarat dasar untuk bertransaksi. Pada kasus ini, umur yang diinput adalah 19, sehingga kondisi tersebut tidak terpenuhi dan program lanjut.<br>
2. Pemilihan Paket Wahana (match-case)<br>
Program menampilkan daftar 5 pilihan wahana beserta harganya, lalu meminta pengguna memasukkan nomor paket (1–5). Nilai yang diinput dicocokkan menggunakan struktur match-case. Karena diinput angka 3, sistem mencocokkan ke "Wahana Motor ATV Ekstrim" dengan harga satuan Rp 120.000. Jika nomor yang diinput di luar 1–5, program akan masuk ke case _ (default), mencetak pesan "Paket wahana tidak valid!", lalu langsung berhenti tanpa memproses input berikutnya.<br>
3. Input Lanjutan (Jumlah Tiket, Member, Promo)<br>
Karena paket yang dipilih valid, program lanjut meminta jumlah tiket, status keanggotaan member, dan status kode promo. Ada validasi if tunggal tambahan: jika jumlah tiket ≤ 0, dicetak peringatan kuota tidak valid — namun karena jumlah tiket yang diinput adalah 5 (bilangan positif), peringatan ini tidak muncul.<br>
4. Validasi Izin Kendali Wahana (if-elif-else dengan operator logika)<br>
Bagian ini mengevaluasi apakah pengunjung boleh mengendalikan wahana secara mandiri, terutama untuk paket ATV. Menggunakan rangkaian kondisi if-elif-elif-elif-elif-else yang menggabungkan pengecekan paket, umur, dan kepemilikan SIM sekaligus dengan operator logika and dan !=. Karena paket yang dipilih adalah 3, umur 19 (≥17), dan sudah punya SIM C, kondisi pertama langsung terpenuhi sehingga dicetak: "Anda sudah dewasa dan boleh mengendarai ATV sendiri." Seluruh kondisi lain di bawahnya otomatis dilewati karena kondisi pertama sudah cocok.<br>
5. Perhitungan Subtotal & Akumulasi Diskon Bertingkat (Multi-IF terpisah)<br>
Subtotal dihitung dari harga satuan dikalikan jumlah tiket (Rp 120.000 × 5 = Rp 600.000). Selanjutnya, empat kondisi if yang berdiri sendiri-sendiri (bukan elif) dievaluasi satu per satu tanpa saling menggugurkan, sehingga diskonnya bisa ditumpuk/diakumulasikan:<br>
Subtotal ≥ Rp 200.000 → tambah diskon 10% (Diskon Belanja Besar)<br>
Berstatus member → tambah diskon 5% (Diskon Member)<br>
Kode promo valid → tambah diskon 15% (Diskon Voucher Promo)<br>
Jumlah tiket ≥ 5 → tambah diskon 5% (Diskon Rombongan)<br>
Karena keempat kondisi tersebut semuanya terpenuhi pada kasus ini, total diskon terakumulasi menjadi 10% + 5% + 15% + 5% = 35%.<br>

6. Evaluasi Kelulusan Audit & Rincian Pembayaran (if-else)<br>
Nominal diskon dihitung dari subtotal dikali persentase diskon (Rp 600.000 × 35% = Rp 210.000), sehingga total bayar menjadi Rp 600.000 − Rp 210.000 = Rp 390.000. Terakhir, digunakan struktur if-else sederhana: jika total bayar lebih dari Rp 300.000, dicetak pesan bonus "Selamat! Anda berhak mendapatkan Souvenir Gratis." Karena Rp 390.000 > Rp 300.000, kondisi ini terpenuhi sehingga pesan bonus tersebut yang tercetak, bukan ucapan terima kasih biasa. Program ditutup dengan menampilkan seluruh rincian pembayaran (subtotal, diskon, total bayar, catatan layanan) lalu mencetak "Program Selesai".<br>
Ringkasan struktur percabangan yang digunakan:<br>
if tunggal — validasi umur minimum & validasi kuota tiket<br>
match-case — pemilihan paket wahana (case 1–5 dan case default _)<br>
if-elif-else dengan operator logika (and, !=) — validasi kelayakan pengendara ATV, satu level tanpa nested-if<br>
Multi-if terpisah (4 blok berdiri sendiri) — akumulasi diskon bertingkat yang bisa ditumpuk<br>
if-else — evaluasi kelulusan audit pembayaran (bonus souvenir vs ucapan terima kasih)
</div>

---
<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">

<h1 align="center"><b>BAB III<br>KESIMPULAN</b></h1>
<p style="text-indent: 40px; text-align: justify; "> Conditional Statement merupakan salah satu struktur kontrol paling fundamental dalam pemrograman, yang memungkinkan sebuah program untuk "mengambil keputusan" dan menentukan blok kode mana yang akan dijalankan berdasarkan hasil evaluasi kondisi logika (True atau False), bukan berjalan secara linear dari atas ke bawah tanpa arah alternatif. Struktur ini dibangun dari tiga elemen utama yang saling melengkapi, yaitu if sebagai pemeriksa kondisi awal, elif sebagai penambah kemungkinan kondisi lain jika kondisi sebelumnya tidak terpenuhi, dan else sebagai blok penutup yang menampung seluruh kemungkinan yang tersisa. Ketiganya dapat dikombinasikan sesuai kebutuhan, mulai dari percabangan sederhana satu kondisi (if saja), dua kemungkinan (if-else), hingga banyak kemungkinan sekaligus (if-elif-else). Dalam proses eksekusinya, sistem selalu memeriksa kondisi secara berurutan dari atas ke bawah, dan begitu ditemukan satu kondisi yang bernilai True, blok kode terkait langsung dijalankan sementara seluruh kondisi lain yang tersisa akan diabaikan (bersifat eksklusif mutual). Karena sifat inilah, urutan penyusunan kondisi menjadi sangat penting dan harus disusun secara logis dan cermat, sebab urutan yang keliru dapat menghasilkan alur eksekusi program yang tidak sesuai dengan yang diharapkan.Untuk membentuk kondisi yang akan dievaluasi, conditional statement memanfaatkan dua kelompok operator, yaitu operator perbandingan (untuk membandingkan nilai) dan operator logika (untuk menggabungkan atau membalik beberapa kondisi sekaligus), sehingga memungkinkan pembuatan aturan pengambilan keputusan yang fleksibel dan kompleks sesuai kebutuhan program. Secara keseluruhan, pemahaman yang kuat terhadap conditional statement menjadi fondasi penting sebelum mempelajari konsep-konsep pemrograman yang lebih lanjut, seperti perulangan (looping) dan fungsi (function), karena hampir seluruh logika pengambilan keputusan dalam program — baik yang sederhana maupun kompleks — pada dasarnya dibangun di atas prinsip percabangan (branching) yang telah dijelaskan dalam teori ini.
</p>
</div>


---
<div style="height:1123px;font-family:'Times New Roman', Times, serif; font-size: 18px;">

<h1 align="center"><b>DAFTAR PUSTAKA</b></h1>

1. Python Software Foundation. (2024). The Python Tutorial — More Control Flow Tools. Diakses dari<br>
https://docs.python.org/3/tutorial/controlflow.html<br>

2. Python Software Foundation. (2024). The Python Tutorial — Compound statements. Diakses dari https://docs.python.org/3/reference/compound_stmts.html<br>

3. W3Schools. (2024). Python Conditions and If statements. Diakses dari<br> 
https://www.w3schools.com/python/python_conditions.asp<br>

4. GeeksforGeeks. (2024). Decision Making in Python (if, if..else, Nested if, if-elif). Diakses dari <br>
https://www.geeksforgeeks.org/decision-making-python-else-nested-elif/<br>

5. Real Python. (2024). Conditional Statements in Python. Diakses dari <br>
https://realpython.com/python-conditional-statements/<br>
</div>